modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/MultiSelectTag.java

```diff
@@ -0,0 +1,337 @@
+		_sourceItems = new ArrayList<>();
+		_valid = true;
+	}
+
+	@Override
+	protected String getHydratedModuleName() {
+		return "frontend-taglib-clay/MultiSelect";
+	}
+
+	@Override
+	protected Map<String, Object> prepareProps(Map<String, Object> props) {
+		props.put("clearAllTitle", _clearAllTitle);
+		props.put("closeButtonAriaLabel", _closeButtonAriaLabel);
+		props.put("disabled", _disabled);
+		props.put("disabledClearAll", _disabledClearAll);
+		props.put("helpText", _helpText);
+		props.put("inputName", _inputName);
+		props.put("inputValue", _inputValue);
+		props.put("isValid", _valid);
+		props.put("label", _label);
+
+		if (_locator != null) {
+			props.put("locator", _locator);
+		}
+
+		props.put("selectedItems", _selectedItems);
+		props.put("sourceItems", _sourceItems);
+
+		return super.prepareProps(props);
+	}
+
+	@Override
+	protected String processCssClasses(Set<String> cssClasses) {
+		cssClasses.add("form-group");
+
+		return super.processCssClasses(cssClasses);
+	}
+
+	@Override
+	protected int processStartTag() throws Exception {
+		super.processStartTag();
+
+		JspWriter jspWriter = pageContext.getOut();
+
+		String id = getId();
+
+		if (_label != null) {
+			jspWriter.write("<label");
+
+			if (Validator.isNotNull(id)) {
+				jspWriter.write(" for=\"");
+				jspWriter.write(id);
+				jspWriter.write("\"");
+			}
+
+			jspWriter.write(">");
+
+			jspWriter.write(_label);
+			jspWriter.write("</label>");
+		}
+
+		jspWriter.write("<div class=\"input-group\"><div class=\"");
+		jspWriter.write("input-group-item\"><div class=\"form-control ");
+		jspWriter.write("form-control-tag-group input-group\"><div class=\"");
+		jspWriter.write("input-group-item\">");
+
+		List<Map<String, String>> selectedItems = getSelectedItems();
+
+		if (!ListUtil.isEmpty(selectedItems)) {
+			for (Map<String, String> selectedItem : selectedItems) {
+				String selectedItemLabel = selectedItem.get("label");
+
+				LabelTag labelTag = new LabelTag();
+
+				labelTag.setDismissible(true);
+				labelTag.setLabel(selectedItemLabel);
+
+				labelTag.doTag(pageContext);
+
+				jspWriter.write("<input type=\"hidden\" name=\"");
+				jspWriter.write(selectedItemLabel);
+				jspWriter.write("\" ");
+				jspWriter.write("value=\"");
+				jspWriter.write(selectedItem.get("value"));
+				jspWriter.write("\" />");
+			}
+		}
+
+		jspWriter.write("<input class=\"form-control-inset\"");
+
+		if (_disabled) {
+			jspWriter.write(" disabled");
+		}
+
+		if (Validator.isNotNull(id)) {
+			super.writeIdAttribute();
+		}
+
+		jspWriter.write(" type=\"text\" /></div>");
+
+		if (!_disabled && !_disabledClearAll && !selectedItems.isEmpty()) {
+			jspWriter.write("<div class=\"input-group-item ");
+			jspWriter.write("input-group-item-shrink\">");
+
+			ButtonTag buttonTag = new ButtonTag();
+
+			buttonTag.setCssClass("component-action");
+			buttonTag.setDisplayType("");
+			buttonTag.setIcon("times-circle");
+
+			if (Validator.isNotNull(_clearAllTitle)) {
```

Comment:
`clearAllTitle` defaults to `Clear All`  [in Clay](https://github.com/liferay/clay/blob/master/packages/clay-multi-select/src/index.tsx#L157) so make sure we set the same default here (probably in `doStartTag` method or in the getter.

Reply:
I've [put that in the getter](https://github.com/liferay-frontend/liferay-portal/pull/528/commits/ab6c891ef7c4f598781dbdec042a518b36db6e8b), but please check if that's correct when you get a chance.

Reply:
This is probably better to be done in `doStartTag` an localize it. And do the same for `closeButtonAriaLabel`